src/webui/api/appcontroller.cpp

```diff
@@ -1159,8 +1160,14 @@ void AppController::getDirectoryContentAction()
         throw APIError(APIErrorType::BadParams, tr("Invalid mode, allowed values: %1").arg(u"all, dirs, files"_s));
     };
 
-    const QStringList dirs = dir.entryList(QDir::NoDotAndDotDot | parseDirectoryContentMode(visibility));
-    setResult(QJsonArray::fromStringList(dirs));
+    QJsonArray ret;
+    QDirIterator it {dirPath, (QDir::NoDotAndDotDot | parseDirectoryContentMode(visibility))};
+    while (it.hasNext())
+    {
+        const QString filePath = it.next();
+        ret.append(filePath);
```

Comment:
I know that Qt containers are implicitly shared. But, IMO, we should follow best c++ practices where possible (as a reinforcing habit). Here, avoid making unnecessary copies.
Aka
```c++
while (it.hasNext())
  ret.append(it.next());
```

Reply:
Done.